libs/product/src/models/product.ts

```diff
@@ -1,11 +1,16 @@
 import { DaffProductImage } from './product-image';
 
+export enum DaffProductTypeEnum {
+	Simple = 'simple',
+	Composite = 'composite'
+}
+
 /**
  * An interface for a product usable by the @daffodil/product library.
  */
 export interface DaffProduct {
 	id: string;
-	__typename?: string;
+	__typename?: DaffProductTypeEnum;
```

Comment:
can we add a new key called `type` as well as `__typename` and then mark `__typename` as deprecated?

Reply:
done